src/EnergyPlus/SimAirServingZones.cc

```diff
@@ -1142,10 +1142,10 @@ namespace SimAirServingZones {
 							}
 						}
 						// added to fix bug issue #5695, if HW coil on outdoor air system, don't lock out during economizing
-						for (OASysNum = 1; OASysNum <= NumOASystems; ++OASysNum) {
-							for (OACompNum = 1; OACompNum <= OutsideAirSys( OASysNum ).NumComponents; ++OACompNum) {
-								CompType = OutsideAirSys( AirSysNum ).ComponentType( OACompNum );
-								if (SameString( CompType, "Coil:Heating:Water" )) {
+						for ( OASysNum = 1; OASysNum <= NumOASystems; ++OASysNum ) {
```

Comment:
@nealkruis @Nigusse This for loop is wrong.  This check should be looking at just one particular OASys, not all of them.  If I'm following correctly, the current code will set `CanBeLockedOutByEcono` to false for every OA system water coil controller in all air loops if it finds even one hot water coil in one OA system anywhere in the model.  The for loop on OASysNum should be removed, and the loop over OACompNum should be moved up around line 1134 and should include a node match check.

Reply:
And why does [line 1136](https://github.com/NREL/EnergyPlus/blob/7c3bbe48305fc421887df2fd5fb75cfb72e4075b/src/EnergyPlus/SimAirServingZones.cc#L1136) only check for "Coil:Cooling:Water:DetailedGeometry" and "Coil:Heating:Water"?  Coil:Cooling:Water can't be locked out?  Other heating coil types?

Reply:
Duh - didn't scroll right enough.  Coil:Cooling:Water is included in line 1136, but what about other non-water coils.  And the comment above seems to indicate that the real fix to #5695 should be to lockout the electric and gas heating coils, not to let the hot water coil operate.  This whole section needs to be rethought carefully - when should OA coils be allowed to run?

Reply:
@mjwitte It seems like there are two issues here:

1. Array bounds hard crash
2. Incorrect implementation of coil lockouts

Depending on the effort of <i>#</i>2, it might fit into this PR. If it's a larger effort, perhaps we should create a new issue and fix it on another branch.

Reply:
Yeah, this PR should just fix he lockout under the original premise that heating coils in the OA system do not get locked out.  Then I'll consider posting a separate issue to see if this should be changed, or more likely, given more user control.

Reply:
@nealkruis So, do you want to modify this to move the check up into the right loop?  Or shall I?

Reply:
@mjwitte I think you have a better idea as to what exactly needs to be done here, but I'm happy to take a stab at it.